stablehlo/reference/Index.h

```diff
@@ -121,6 +121,8 @@ using Index = Sizes;
 class IndexSpaceIterator {
  public:
   /// \name Constructor
+  IndexSpaceIterator(Sizes shape) : shape_(shape) { index_ = std::nullopt; }
```

Comment:
Nit: Would it be possible to use a default value of `std::nullopt` for the `index` in the constructor below? I personally think that's cleaner but up to you.

Reply:
Done.